src/forest.rs

```diff
@@ -243,10 +241,324 @@ impl Forest {
 
         mark_dirty_recursive(&mut self.nodes, &self.parents, node);
     }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Forest, NodeData};
+    use crate::geometry::Size;
+    use crate::node::{MeasureFunc, NodeId};
+    use crate::style::FlexboxLayout;
+    use crate::sys::new_vec_with_capacity;
+
+    fn assert_forest_size(forest: &Forest, size: usize) {
```

Comment:
If you have 1 parent with 2 children in the tree would the length of the vectors return:
```
Nodes: 3
Children: 2
Parents: 1
```
Or do they report their capacity?

Reply:
So, this is one of the quirks of this data structure, actually the vectors must always have the same length.

- `nodes` with 3 elements containing their `NodeData` indexed by `NodeId`
- `parents` contain one entry for each node, also indexed by `NodeId`, and inside each entry there's another vector storing all of the elements' parents. That means that every children would have an entry here ponting to their single parent. 
- `children` works the same as the `parent`, just flipped accordingly. 
- Those inner vectors are not indexed by `NodeId`, just stored as they come.

In the case you've presented: 
`children[parent_id]` would have len == 2
`children[first_child_id]` would by empty
`children[second_child_id]` would by empty

`parents[parent_id]` would by empty
`parents[first_child_id]` would have len == 1
`parents[second_child_id]` would have len == 1

Maybe it would make sense to just create a new function in `Forest` that reports the node count?

Reply:
I would like that method :) Can you also leave a comment documenting this weird behavior?

Reply:
I've create a `len` function and placed some comments indicating that all of them must always have the same length.
For the tests I've kept this `assert_forest_size` which I think is a more robust way of asserting the forest consistency.

Reply:
Excellent, I like those changes.